its/src/test/java/com/sonar/it/shared/TestUtils.java

```diff
@@ -82,7 +82,27 @@ public static ScannerForMSBuild newEndStep(Path projectDir) {
       .addArgument("end");
   }
 
-  public static Build<ScannerForMSBuild> newScanner(Path projectDir) {
+  public static  ScannerForMSBuild createStartStep(String projectName, Path projectDir) {
+    return createStartStep(projectName, projectDir, "");
+  }
+
+  public static  ScannerForMSBuild createStartStep(String projectName, Path projectDir, String subProjectName) {
+    return TestUtils.newScanner(projectDir)
+      .addArgument("begin")
+      .setProjectKey(projectName)
+      .setProjectName(projectName)
+      .setProjectVersion("1.0")
+      .setProperty("sonar.projectBaseDir", getProjectBaseDir(projectDir, subProjectName));
+  }
+
+  private static String getProjectBaseDir(Path projectDir, String subProjectName){
+    if (subProjectName.isEmpty()) {
+      return projectDir.toString();
+    }
+    return projectDir.toString() + File.separator + subProjectName;
```

Comment:
_nitpick (feel free to ignore)_: a more elegant solution would be to call `projectDir.resolve(subProjectName)` and `toString` the result, see https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#resolve-java.lang.String-

Reply:
That's actually great. I was looking for something similar but I didn't knew how to do this in Java.